crates/tabby-common/src/index/mod.rs

```diff
@@ -13,6 +13,26 @@ use tantivy::{
     DateTime, Term,
 };
 
+/// On a high level, the index schema is structured as follows:
+///
+///           +----------------+
+///           |     corpus     | <--- A group of documents, each document has a unique
+///           +----------------+      identifier (id) within the corpus.
+///                   |
+///                   v
+///           +----------------+
+///           |    document    | <--- A document is a group of chunks, each document has a
+///           |      (id)      |      unique identifier (id) across corpus, document's
+///           +----------------+      attributes are stored but not indexed.
+///                   |
+///                   v
+///           +----------------+
+///           |     chunk      | <--- Each chunk has a unique identifier (chunk_id) within
+///           |   (chunk_id)   |      the document. Chunk is the unit being retrieved during
+///           +----------------+      the search process.
+///
+/// Across the corpus, there is a concept of source_id, which identifies a group of documents.
+/// It is usually used to identify the source of the document, such as a Github connection or a web document crawl.
```

Comment:
@xxs-wallace fyi

Reply:
Since source id is not composed by git url, could there be several document for one github repository?

Reply:
yes - and it's the indended behavior.